File: frontend/utils/api_client.py
import requests
from typing import List, Dict, Optional
import streamlit as st

# API configuration
API_BASE_URL = "http://localhost:8000"

class APIClient:
    def __init__(self, base_url: str = API_BASE_URL):
        self.base_url = base_url
    
    def _make_request(self, method: str, endpoint: str, **kwargs):
        """Make HTTP request to API"""
        url = f"{self.base_url}{endpoint}"
        try:
            response = requests.request(method, url, **kwargs)
            response.raise_for_status()
            return response.json()
        except requests.exceptions.RequestException as e:
            st.error(f"API Error: {e}")
            return None
    
    # Categories
    def get_categories(self) -> List[Dict]:
        """Get all categories"""
        return self._make_request("GET", "/categories/")
    
    def create_category(self, name: str) -> Dict:
        """Create a new category"""
        return self._make_request("POST", "/categories/", params={"name": name})
    
    # Items
    def get_items(self) -> List[Dict]:
        """Get all items"""
        return self._make_request("GET", "/items/")
    
    def get_item(self, item_id: int) -> Dict:
        """Get specific item"""
        return self._make_request("GET", f"/items/{item_id}")
    
    def create_item(self, name: str, unit: str, restock_threshold: float, category_id: int) -> Dict:
        """Create a new item"""
        return self._make_request("POST", "/items/", params={
            "name": name,
            "unit": unit,
            "restock_threshold": restock_threshold,
            "category_id": category_id
        })
    
    def update_item(self, item_id: int, **kwargs) -> Dict:
        """Update an item"""
        return self._make_request("PUT", f"/items/{item_id}", params=kwargs)
    
    def delete_item(self, item_id: int) -> Dict:
        """Delete an item"""
        return self._make_request("DELETE", f"/items/{item_id}")
    
    def get_items_by_category(self, category_id: int) -> List[Dict]:
        """Get items by category"""
        return self._make_request("GET", f"/items/category/{category_id}")
    
    # Stock Records
    def get_stock_records(self) -> List[Dict]:
        """Get all stock records"""
        return self._make_request("GET", "/stocks/")
    
    def log_stock(self, item_id: int, quantity: float, notes: Optional[str] = None) -> Dict:
        """Log a stock count"""
        params = {"item_id": item_id, "quantity": quantity}
        if notes:
            params["notes"] = notes
        return self._make_request("POST", "/stocks/", params=params)
    
    def get_stock_records_for_item(self, item_id: int) -> List[Dict]:
        """Get stock records for specific item"""
        return self._make_request("GET", f"/stocks/item/{item_id}")
    
    # Restock Records
    def get_restock_records(self) -> List[Dict]:
        """Get all restock records"""
        return self._make_request("GET", "/restocks/")
    
    def log_restock(self, item_id: int, restock_amount: float, supplier: Optional[str] = None, notes: Optional[str] = None) -> Dict:
        """Log a restock"""
        params = {"item_id": item_id, "restock_amount": restock_amount}
        if supplier:
            params["supplier"] = supplier
        if notes:
            params["notes"] = notes
        return self._make_request("POST", "/restocks/", params=params)
    
    def get_restock_records_for_item(self, item_id: int) -> List[Dict]:
        """Get restock records for specific item"""
        return self._make_request("GET", f"/restocks/item/{item_id}")

# Create a global API client instance
api_client = APIClient() 

File: frontend_v2/src/utils/apiClient.js
import axios from 'axios';

const API_BASE_URL = 'http://localhost:8000';

class APIClient {
  constructor() {
    this.api = axios.create({
      baseURL: API_BASE_URL,
      timeout: 10000,
    });
  }

  // Helper method to handle API calls
  async makeRequest(method, endpoint, params = {}) {
    try {
      const response = await this.api.request({
        method,
        url: endpoint,
        params,
      });
      return response.data;
    } catch (error) {
      console.error('API Error:', error);
      throw error;
    }
  }

  // Categories
  async getCategories() {
    return this.makeRequest('GET', '/categories/');
  }

  async createCategory(name) {
    return this.makeRequest('POST', '/categories/', { name });
  }

  async updateCategory(categoryId, name, description = null) {
    const params = { name };
    if (description !== null) {
      params.description = description;
    }
    return this.makeRequest('PUT', `/categories/${categoryId}`, params);
  }

  async deleteCategory(categoryId) {
    return this.makeRequest('DELETE', `/categories/${categoryId}`);
  }

  // Items
  async getItems() {
    return this.makeRequest('GET', '/items/');
  }

  async getItem(itemId) {
    return this.makeRequest('GET', `/items/${itemId}`);
  }

  async createItem(name, unit, restockThreshold, categoryId) {
    return this.makeRequest('POST', '/items/', {
      name,
      unit,
      restock_threshold: restockThreshold,
      category_id: categoryId,
    });
  }

  async updateItem(itemId, updates) {
    return this.makeRequest('PUT', `/items/${itemId}`, updates);
  }

  async deleteItem(itemId) {
    return this.makeRequest('DELETE', `/items/${itemId}`);
  }

  async getItemsByCategory(categoryId) {
    return this.makeRequest('GET', `/items/category/${categoryId}`);
  }

  // Stock Records
  async getStockRecords() {
    return this.makeRequest('GET', '/stock/');
  }

  async logStock(itemId, quantity, notes = null, staffName = null) {
    const params = { item_id: itemId, quantity };
    if (notes) params.notes = notes;
    if (staffName) params.staff_name = staffName;
    return this.makeRequest('POST', '/stock/', params);
  }

  async getStockRecordsForItem(itemId) {
    return this.makeRequest('GET', `/stock/item/${itemId}`);
  }

  // Restock Records
  async getRestockRecords() {
    return this.makeRequest('GET', '/restocks/');
  }

  async logRestock(itemId, restockAmount, supplier = null, notes = null) {
    const params = { item_id: itemId, restock_amount: restockAmount };
    if (supplier) params.supplier = supplier;
    if (notes) params.notes = notes;
    return this.makeRequest('POST', '/restocks/', params);
  }

  async getRestockRecordsForItem(itemId) {
    return this.makeRequest('GET', `/restocks/item/${itemId}`);
  }
}

export default new APIClient(); 